src/Chapter3.hs

```diff
@@ -970,6 +1114,27 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Weekday
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday 
+  deriving (Show, Eq, Bounded, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Saturday  = True
+isWeekend Sunday    = True
+isWeekend _         = False
+
+nextDay :: Weekday -> Weekday
+nextDay weekday = if weekday == maxBound then minBound else succ weekday 
+
+daysToParty :: Weekday -> Int
+daysToParty weekday = mod ((fromEnum Friday) + daysInWeek - (fromEnum weekday)) daysInWeek
+  where daysInWeek = (fromEnum (maxBound::Weekday)) - (fromEnum (minBound::Weekday)) + 1
```

Comment:
I changed the code not to use my definition of Weekday where Friday is the last day of the week. I change Weekday s.t. Sunday is the last day of the week. I changed the definition of daysToParty not to depend on the last day of the week.